File: Recursion/523-continuous-subarray-sum/continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        
        unordered_map<int,int> mp;
        mp[0]=-1;
        long long prefix=0;
        for(int i=0;i<nums.size();i++){
            prefix+=nums[i];
            int mod = prefix%k;
            if(mp.count(mod)){
                if(i-mp[mod]>=2){
                    return true;
                }
            }
            else{
                mp[mod]=i;
            }
        }
        return false;
    }
};

File: Linked_List/523-continuous-subarray-sum/continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {

        HashMap<Integer,Integer> map = new HashMap<>();

        map.put(0,-1);

        if(nums.length<=1) return false;

        int sum =0;

        for(int i=0;i<nums.length;i++){

            sum+=nums[i];

            int ava = sum%k;

            if(map.containsKey(ava)){

                if(i-map.get(ava)>=2) return true;
            }

            else {
                map.put(ava,i);
            }
        }

        return false;
        
    }
}